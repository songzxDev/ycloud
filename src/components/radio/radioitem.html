<label class="y-radio-item" data-bind="css:{'y-radio-item-checked': checked}">
  <span class="y-radio-item--inner-warpper">
    <span class="y-radio-item-inner">
      <input type="radio" data-bind="attr:{name:radioname},checked: checked, event:{change:change }">
    </span>
  </span>
  <span class="y-radio-label" data-bind="text:value.label"></span>
</label>
