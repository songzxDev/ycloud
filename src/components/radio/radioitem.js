import template from './radioitem.html'
import ko from 'knockout'
function init (params) {
  this.value = params.value
  this.currentValue = params.currentValue
  this.radioname = params.radioname
  this.checked = ko.computed(() => {
    // todo 默认是value/label 后续要改成其他自定义格式
    return this.currentValue() === this.value.value
  })
  this.change = (data) => {
    this.currentValue(this.value)
  }
}
export default {
  name: 'radio-item',
  init,
  template
}
