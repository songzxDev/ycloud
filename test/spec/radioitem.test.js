import ko from 'knockout'
import radioitem from '../../src/components/radio/radioitem.js'
import $ from 'jquery'
describe('y-radioitem组件', () => {
  const PREFIX = 'y-'
  ko.components.register(PREFIX + radioitem.name, {
    viewModel: radioitem.init,
    template: radioitem.template,
    synchronous: true
  })
  ko.cleanNode(document.getElementsByTagName('body')[0])
  let vm = {
    title: ko.observable('test'),
    item: {value: 1,label:'男'},
    currentValue: ko.observable(1),
    radioname: 'y-radio-' + 123
  }
  document.body.innerHTML = '<y-radio-item params="radioname: radioname, value: item, currentValue: currentValue"></y-radio-item>'
  ko.applyBindings(vm)
  var el = $('body')
  it("组件注册测试", () => {
    expect(radioitem).not.toBeNull();
    expect(ko.components.isRegistered(PREFIX + radioitem.name)).toBe(true);
  })
  it('生成radioitem测试', () => {
    expect($('body').find('.y-radio-item').length > 0).toEqual(true)
  })
  it('radioitem默认值设置', () => {
    expect($('body').find('.y-radio-item').hasClass('y-radio-item-checked')).toEqual(true)
  })
  it('快照', () => {
    expect($('body').get(0)).toMatchSnapshot()
  })
})
