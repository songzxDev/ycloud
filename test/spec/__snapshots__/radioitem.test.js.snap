// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`y-radioitem组件 快照 1`] = `
<body>
  <y-radio-item
    params="radioname: radioname, value: item, currentValue: currentValue"
  >
    <label
      class="y-radio-item y-radio-item-checked"
      data-bind="css:{'y-radio-item-checked': checked}"
    >
      
  
      <span
        class="y-radio-item--inner-warpper"
      >
        
    
        <span
          class="y-radio-item-inner"
        >
          
      
          <input
            data-bind="attr:{name:radioname},checked: checked, event:{change:change }"
            name="y-radio-123"
            type="radio"
          />
          
    
        </span>
        
  
      </span>
      
  
      <span
        class="y-radio-label"
        data-bind="text:value.label"
      >
        男
      </span>
      

    </label>
    

  </y-radio-item>
</body>
`;
